lms/static/js/ccx/schedule.js

```diff
@@ -343,13 +376,22 @@ var edx = edx || {};
 	      children.hide();
 	    }
 	    else {
-				$(this).attr('aria-expanded', 'true');
-	      $(this).find(".fa-caret-right").removeClass('fa-caret-right').addClass('fa-caret-down');
-	      row.removeClass('collapsed').addClass('expanded');
-	      children.filter('.collapsed').each(function() {
-		children = children.not(self.get_children(this));
-	      });
-	      children.show();
+                $(this).attr('aria-expanded', 'true');
+                $(this).find(".fa-caret-right").removeClass('fa-caret-right').addClass('fa-caret-down');
+                row.removeClass('collapsed').addClass('expanded');
+                var depth = $(row).data('depth');
+                var $childNodes = children.filter('.collapsed');
+                if ($childNodes.length <= 0) {
+                  children.show();
+                } else {
+									// this will expand units.
```

Comment:
I will fix this in refactoring , currently there is mix use of tabs and spaces in this file.